### PR TITLE
fix: ensure `predict_proba` is a probability or `None`

### PR DESCRIPTION
**Pull request by tachyonicClock**
See https://github.com/adaptive-machine-learning/backlog/issues/26 or https://github.com/adaptive-machine-learning/backlog/issues/27

### CODE REVIEW

**Comment by hmgomes** [comment]
HI @tachyonicClock , we need to do something about the tests to allow for the None predictions. Should we somehow identify whether it is a None prediction before the model is trained on anything or a None prediction afterwards (some sort of error or abstaining from voting as described in [https://github.com/adaptive-machine-learning/backlog/issues/27](https://github.com/adaptive-machine-learning/backlog/issues/27))

We have the functionality to explicitly allowing abstaining from predicting (see bellow), and it might be applicable for a separate issue, but should we keep track of the abstentions? 
https://github.com/adaptive-machine-learning/CapyMOA/blob/9e66b3912a4bd1b361c0e806867367f9bc483211/src/capymoa/evaluation/evaluation.py#L181

File: src/capymoa/base/_classifier.py [hunk 1]
```diff
@@ -241,13 +240,12 @@ def cli_help(self):
...
-        return Utils.maxIndex(
-            self.moa_learner.getVotesForInstance(instance.java_instance)
-        )
-
-    def predict_proba(self, instance):
-        return self.moa_learner.getVotesForInstance(instance.java_instance)
+    def predict_proba(self, instance) -> Optional[LabelProbabilities]:
+        votes = np.array(self.moa_learner.getVotesForInstance(instance.java_instance))
+        total = sum(votes)
+        if total <= 1e-2 or np.isnan(total) or np.isinf(total):
```
Comment (tachyonicClock) [inline src/capymoa/base/_classifier.py:246]: @hmgomes I believe this allows moa models to abstain
